jax/_src/array.py

```diff
@@ -12,12 +12,11 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+
 import abc
```

Comment:
Are we sure we want to rename this file right now?

I was planning to do it while moving the entire thing in experimental/array.py here and sharding.py too.

Reply:
Currently, now there are 2 types of imports which is confusing. Can we just do the rename when the concrete class moves too?

Reply:
Do you mean we should define `jax.Array` within `jax._src.numpy.ndarray`?

Reply:
Yes, as a stopgap. If you want to.. maybe move the entire Array implementation here with this rename?

But I guess that will touch a lot more files :)

And then you can add a big comment, telling what to move to jax._src.array when you/I do the move.

Reply:
I think I'll put it in `jax._src.typing` for now instead.